internal/rpcserver/rpcserver.go

```diff
@@ -5849,6 +5849,39 @@ func (s *Server) authMAC(dst, auth []byte) []byte {
 	return dst
 }
 
+// checkAuthHeader checks the HTTP Basic authentication string by comparing
+// it with the already generated hash.
+//
+// The first bool return value signifies auth success (true if successful) and
+// the second bool return value specifies whether the user can change the state
+// of the server (true) or whether the user is limited (false).
+func (s *Server) checkAuthHeader(auth, remoteAddr string) (bool, bool) {
```

Comment:
This doesn't really do what it's named since it's used for both the `authenticate` RPC and HTTP basic authentication paths.  I would say it should be named `checkAuthMAC`.

Reply:
Renamed.